app/assets/javascripts/incidents/batch_export.coffee

```diff
@@ -0,0 +1,29 @@
+handleExporting = ->
+  inSelectMode = $('table.incidents th.batch-export').is ':visible'
+  if inSelectMode
+    ids = $('.batch-export input:checked').map ->
+      $(this).data 'id'
+    .toArray()
+    alreadyExported= $('.batch-export input:checked').filter ->
+      $(this).data 'exported'
+    if alreadyExported.length > 0
+      if alreadyExported.length == 1
+        warning = '1 selected incident has already been exported.'
+      else warning = "#{alreadyExported.length} selected incidents have already been exported."
+      warning += "\nPlease doublecheck that you would like to re-export this incident."
+      unless confirm warning
+        return false
+     window.location.href = 'incidents/batch_export?' + $.param(ids: ids)
+  else
+    $('table.incidents .batch-export').show()
+    $(this).text 'Select incidents to export...'
+
+handleIncidentSelected = ->
+  selectedIncidentCount = $('.batch-export input:checked').length
+  if selectedIncidentCount > 0
+    $('.batch-export button').text "Generate XML export (#{selectedIncidentCount} selected)"
+  else $('.batch-export button').text 'Select incidents to export...'
```

Comment:
It's really jarring not to have an `end`. It's like "Let's make this a little more like ruby but _not all the way_

Reply:
I really only describe Coffeescript as "like Ruby" to people in the office. That wasn't the intent of the language.

This particular feature, in any event, I might describe as Python-ish. The indentation is what signifies the end of a block, not any particular keyword.